File: CCIP/CCIP/EmbeddedTabBarControllerSegue.m
//
//  EmbeddedTabBarControllerSegue.m
//  CCIP
//
//  Created by 腹黒い茶 on 2016/07/30.
//  Copyright © 2016年 CPRTeam. All rights reserved.
//

#import "EmbeddedTabBarControllerSegue.h"

@implementation EmbeddedTabBarControllerSegue

- (void)perform {
    __block UIView *sourceView = self.sourceViewController.view;
    __block UIView *destinationView = self.destinationViewController.view;
    CGRect frame = sourceView.frame;
    //frame.size.height -= [self.sourceViewController.bottomLayoutGuide length];
    frame.size.height += 15;
    [self.sourceViewController presentViewController:self.destinationViewController
                                            animated:YES
                                          completion:^{
                                              destinationView.superview.frame = frame;
                                              [destinationView.superview setClipsToBounds:YES];
                                          }];
}

@end

File: CCIP/CCIP/CustomSegue/EmbeddedTabBarControllerSegue.swift
//
//  EmbeddedTabBarControllerSegue.swift
//  OPass
//
//  Created by 腹黒い茶 on 2018/10/7.
//  Copyright © 2018 OPass. All rights reserved.
//

import Foundation
import UIKit

class EmbeddedTabBarControllerSegue : UIStoryboardSegue {
    override func perform() {
        let sourceView : UIView = self.source.view
        let destinationView : UIView = self.destination.view
        var frame : CGRect = sourceView.frame
        frame.size.height -= self.source.view.safeAreaInsets.bottom
        self.source.present(self.destination, animated: true) {
            destinationView.superview!.frame = frame
            destinationView.superview!.clipsToBounds = true
        }
    }
}
